gate-web/src/main/groovy/com/netflix/spinnaker/gate/filters/FiatSessionFilter.groovy

```diff
@@ -52,29 +56,36 @@ class FiatSessionFilter implements Filter {
    */
   @Override
   void doFilter(ServletRequest request, ServletResponse response, FilterChain chain) throws IOException, ServletException {
-    if (configProps.enabled) {
+    if (fiatConfig.enabled && this.enabled) {
       String user = AuthenticatedRequest.getSpinnakerUser().orElse(null)
+      log.debug("Fiat session filter - found user: ${user}")
       if (permissionEvaluator.getPermission(user) == null) {
         HttpServletRequest httpReq = (HttpServletRequest) request
         HttpSession session = httpReq.getSession(false)
         if (session != null) {
           log.info("Invalidating user '{}' session '{}' because Fiat permission was not found.",
-          value("user", user), value("session", session))
+                   value("user", user),
+                   value("session", session))
           session.invalidate()
           SecurityContextHolder.clearContext()
         }
       }
+    } else {
+      if (log.isDebugEnabled()) {
```

Comment:
Is this common to not log debug logs if debug mode is not enabled?

Reply:
The check is implicit in the `.debug()` call, but making it explicit skips the JVM from resolving all of the arguments before the method is actually called (and if disabled, doing nothing with those arguments).

So in this case, by first checking if it's enabled, I'm skipping the JVM creating and concatenating the debug message string. Not much saving in this case, but if the debug message you were constructing were much more expensive to create, then it could potentially have a big runtime impact.